services/QuillCMS/app/workers/rematch_response_worker.rb

```diff
@@ -21,13 +21,17 @@ class RematchResponseWorker
 
   class LambdaHTTPError < StandardError; end
 
-  def perform(response_id, question_type, question_hash, reference_response_ids)
+  def perform(response_id, question_type, question_hash, reference_response_ids, options)
     response = Response.find_by(id: response_id)
     return unless response
 
     reference_responses = Response.where(id: reference_response_ids).to_a
 
     rematch_response(response, question_type, question_hash, reference_responses)
+
+    return unless options['fire_pusher_alert'] && options['question_key']
+
+    RematchingFinished.run(options['question_key'])
```

Comment:
Minor note: the way that this is structured, if there's an error in `RematchingFinished.run` (like, say, Pusher API is down or something), that'll cause this entire job to be put into the retry queue.  So the `rematch_response` function will get called on this specific response again.

The way around this would be to put `RematchingFinished` logic into its own independent worker, and just have this job enqueue that job when it needs to.  That said, I think that the rematch for a single response should be fast enough that if we end up re-running one of them because of a Pusher error, it wouldn't be the end of the world, so it may not be worth refactoring this from a practical perspective even if it would be "better".

Reply:
Thanks for flagging, @anathomical. I agree that I think repeating one rematch is pretty low-stakes, but I will keep that approach in mind for the future.

Reply:
Thanks, @anathomical. I agree that having one rematch job rerun is probably pretty low-stakes, but since it's an easy change I'll go ahead and make it.